src/assisted_test_infra/test_infra/controllers/iscsi_target.py

```diff
@@ -0,0 +1,160 @@
+
+    def __str__(self):
+        return f"iqn.{self.year_month}.{self.base_domain}:{self.unique_name}"
+
+
+class Receiver:
+
+    def __init__(self, fn_callback, **defaults_kwargs):
+        self.fn_callback = fn_callback
+        self.defaults_kwargs = defaults_kwargs
+
+    def execute(self, *args, **kwargs):
+        self.defaults_kwargs.update(kwargs)
+        log.info(f"iscsi_target exec: {str(args)} {str(kwargs)}")
+        self.fn_callback(*args, **kwargs)
+
+
+class IscsiTargetConfig:
+    def __init__(self, disk_name: str, disk_size_gb: int, iqn: Iqn, remote_iqn: Iqn, iso_disk_copy: str,
+                 servers: list[str]):
+        self._disk_name = disk_name
+        self._disk_size = disk_size_gb
+        self._iqn = iqn
+        self._data_file = iso_disk_copy
+        self._remote_iqn = remote_iqn
+        self._servers = servers
+
+    @property
+    def disk_name(self):
+        return self._disk_name
+
+    @property
+    def disk_size(self):
+        return self._disk_size
+
+    @property
+    def iqn(self):
+        return self._iqn
+
+    @property
+    def data_file(self):
+        return self._data_file
+
+    @property
+    def remote_iqn(self):
+        return self._remote_iqn
+
+    @property
+    def servers(self):
+        return self._servers
+
+
+class IscsiTargetController:
+
+    def __new__(cls):
+        # Make The controller singleton class because managed by hypervisor as a single service
+        if not hasattr(cls, 'instance'):
+            log.info('Creating singleton IscsiTargetController instance')
+            cls.instance = super(IscsiTargetController, cls).__new__(cls)
+        return cls.instance
+
+    def __init__(self):
+        self.builder = IscsiBuilder()
+        self._target_configs = []
+
+    def create_target_server(self, config: IscsiTargetConfig, clear_config=False) -> None:
+        if clear_config:
+            self.builder.clear_iscsi_target_config()
+        # append target configs to delete on cleanup
+        self._target_configs.append(config)
+
+        self.builder.create_file_disk(config.disk_name, config.disk_size)
+        self.builder.create_file_io(config.disk_name, config.disk_size)
+        self.builder.create_iqn(config.iqn)
+        self.builder.create_lun(config.iqn, config.disk_name)
+        self.builder.create_portal(config.iqn, config.servers)
+        self.builder.create_access_list(config.iqn, config.remote_iqn)
+        self.builder.save_config()
+        self.builder.copy_to_created_disk(config.disk_name, config.data_file)
+
+    def clean_target_server(self) -> None:
+        for config in self._target_configs:
+            # Remove resource file created
+            self.builder.remove_file_disk(config.disk_name)
+        self.builder.clear_iscsi_target_config()
+
+
+class IscsiBuilder:
+    """iSCSI builder implements commands on the receiver (ssh runner).
+
+    The builder can run on any machine with targetcli installed
```

Comment:
I feel that we try to re-implement ansible, it would be nice to think about it in the future